docs/upgrade.md

```diff
@@ -88,6 +88,22 @@ process, for example:
     dpkg -i matrix-synapse-py3_1.3.0+stretch1_amd64.deb
     ```
 
+# Upgrading to v1.74.0
+
+## Unicode support in user search
+
+This version introduces optional support for an [improved user search dealing with Unicode characters](https://github.com/matrix-org/synapse/pull/14464).
+
+If you want to take advantage of this feature you need to install PyICU,
+the ICU native dependency and its development headers
+so that PyICU can build since no prebuilt wheels are available.
+
+You can follow [the PyICU documentation](https://pypi.org/project/PyICU/) to do so,
+and then do `pip install matrix-synapse[icu]` for a PyPI install.
```

Comment:
Shouldn't this have been `matrix-synapse[user-search]` instead of `matrix-synapse[icu]`? The latter did not work for me because `user-search` is the name of the extra in the code (see [here](https://github.com/matrix-org/synapse/pull/14464/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R234-R237)) and using that did work for me

Reply:
Yes, good spot. https://github.com/matrix-org/synapse/pull/14797